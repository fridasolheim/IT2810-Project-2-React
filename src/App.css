.App > * {
    margin-top: 0px;
    /*-webkit-flex: 1 1 0;
    -ms-flex: 1 1 0;
    flex: 1 1 0;*/
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: left;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}
.Content{
  display: flex;
  padding-left: 30px;
  padding-right: 30px;
  /*max-width: 45rem;  /*erstatt med max-width for responsiv(men stygg) løsning
  height: 25rem;  erstatt med max-height for responsiv(men stygg) løsning*/
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;  
}

.subContent{
  display: flex;
  flex-direction: row;
  vertical-align: middle;
  justify-content: space-evenly;
  align-items: center;

}

@media screen and (max-width: 500px){
  .Content {
    padding: 10px 10px;
    font-size: 1em;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
  .container1{
    flex-direction: column;
    align-items: center;
  }
  .container2{
    flex-direction: column;
    align-items: center;
  }
  /*.firstHeader{
    size: 9px !important;
  }
  .secondHeader{
    size: 9px !important;
  }*/
  .buttonStyle{
    height: 20px;
    line-height: 20px;
    width: 75px;
    margin-bottom: 10px;
    margin-top: 10px;
    font-size: 9px;
    align-items: center !important;
  }
  .subContent{
    flex-direction: column;
  }
  .checkboxesStyle{
    align-items: center;
  }
  
  .tabs{
    align-items: center;
  }
}
.mediaContent{
  width: 60%;
  height: 60%;
  flex-direction: row;
}
.image1{
  max-width: 1400vp;
  max-height: 1400vp;
  width: 100%; 
  height: auto;
  align-items: right;
  padding:15px;
}
.tabs{
  max-width: 70%;
}
.break{
  flex-basis: 100%;
}

.text1{
  text-align: left;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  width: 30vp;
  height: 30vp;
  padding:10px;
  /*max-width: 10rem;*/
}

.Header{
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 170px;
  width: 100vp;
  line-height: 35px;
  text-align: center;
  align-items: center;
  align-content: center;
  padding-top: 20px;
}

.firstHeader{
  font-family: Georgia, 'Times New Roman', Times, serif;
  letter-spacing: 2px; 
  color: white;
}
h2{
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  letter-spacing: 1px;
}
.secondHeader{
  font-family: Georgia, 'Times New Roman', Times, serif;
  font-style: italic;
  color: black;
  background: #ccc;
  letter-spacing: 3px; 
  justify-content: space-around;
  color: white;
  font-weight: 200;
}
.container1{
  display: flex;
  justify-content: space-around;
  align-content: left;
  margin-right: 10px;
  margin-left: 5px;
  margin-top: 5px;
}

.container2{
  display: flex;
  justify-content: space-evenly;
  align-content: center;
  padding-left: 20%;
  padding-right: 20%;
  margin-top: 10px;
}

.checkBoxes{
  display: flex;
  justify-content: space-around;
  border: 2px solid black;
  margin-right: 20px;
  margin-left: 20px;  
}

.checkboxesStyle{
  display: flex;
  align-items: left !important;
  flex-direction: column;
  vertical-align: middle;
  float: left;
  margin-bottom: 5px;
}

.tab-list {
  border-bottom: 1px solid #ccc;
  padding-left: 20px;
  padding-right: 20px;
}

.tab-list-item {
  display: inline-block;
  list-style: none;
  margin-bottom: -1px;
  padding: 0.5rem 0.75rem;
  cursor: pointer;
}

.tab-list-active {
  background-color: white;
  border: solid #ccc;
  border-width: 1px 1px 0px 1px;
  text-decoration: black;
  text-decoration-style: solid;
}

.buttonStyle{
  background-color: #282c34;
  height: 40px;
  line-height: 40px;
  width: 120px;
  margin-bottom: 20px;
  margin-top: 20px;
  vertical-align: middle;
  font-size: 14px;
  font-weight: 600;
  color: white;
  text-align: center;
  align-content: center;
}
.buttonStyle:hover{
  /*font-style: italic;*/
  font-weight: 700;
  color: #282c34;
  background-color: white;
  box-shadow: 0px 0px 0px 3px black;
  cursor: pointer;
}
.headerSubmit{
  background-color: white;
  height: 20px;
  line-height: 20px;
  width: 144px;
  vertical-align: middle;
  font-size: 10px;
  font-weight: 600;
  color: #282c34;
  text-align: center;
  align-content: center;
  cursor: pointer;
}
.inputName{
  font-family: Georgia, 'Times New Roman', Times, serif;
  font-style: italic;
  text-align: center;
}

.headerForm{
  display: flex;
  padding-bottom: 40px;
  flex-direction: row;
}

.pic{
  width: 100vp;
  max-width:60px !important;
  max-height: 60px !important;
}

.textContent{
  width: 100vp;
  max-width: 50rem;
}

