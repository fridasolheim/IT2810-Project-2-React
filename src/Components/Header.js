import React, {Component} from 'react';
import '../App.css';

class Header extends Component{
    constructor(props){
        super(props);
        this.state = {
            value: ' ',
            shouldHide: true,
            headerText: " ",
        };
        
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    };
    
    handleChange(event){
        this.setState({value: event.target.value});
    };
    
    handleSubmit(event){
        //alert('A name was submitted: ' + this.state.value);
        this.setState({shouldHide: false});
        sessionStorage.setItem('userValue', this.state.value.toUpperCase());
        event.preventDefault();
        
    };
    
    render(){
        return (
        <div className='Header'>
            <h1 className="h1-header">WELCOME {sessionStorage.getItem('userValue')}</h1>
            <h5 className="h5-header">TO OUR RANDOM ART GENERATOR</h5>
            <div className="headerForm">
                <form style={this.state.shouldHide ? {} : { display: 'none' }}>
                    <div>
                        <input type="text" className="inputName" onChange={this.handleChange} placeholder="What's your name"/>
                        <div onClick={this.handleSubmit} className="headerSubmit">SUBMIT</div>
                    </div>
                </form>
            </div>
        </div>
        );
    };
};

export default Header;