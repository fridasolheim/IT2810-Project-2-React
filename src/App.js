import React,{Component} from 'react';
import kiwi from './kiwi.svg';
import './App.css';
import Header from './Components/Header';
import Tabs from './Components/Tabs';
import Page from './Components/Page';


class App extends Component {
   
  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);

    this.state = {
      chosenPicture: 1,
      chosenSound: 4,
      chosenText: 7,
      activeTab:1,

      pictures: [
        {
          id: 1,
          title: 'Elephant',
          completed: true,
        },
        {
          id: 2,
          title: 'Giraffe',
          completed: false,
        },
        {
          id: 3,
          title: 'Dog',
          completed: false,
        }], 
      sounds:[
        {
          id: 4,
          title: 'Techno',
          completed: true,
        },
        {
          id: 5,
          title: 'Jazz',
          completed: false,
        },
        {
          id: 6,
          title: 'Telephone',
          completed: false,
        }],
      text:[
        {
          id: 7,
          title: 'Poem',
          completed: true,
        },
        {
          id: 8,
          title: 'Novel',
          completed: false,
        },
        {
          id: 9,
          title: 'Lyrics',
          completed: false,
        }
    ]}
}

  handleSave = () => {
    //const {chosenPicture, chosenSound, chosenText} = this.state;
    localStorage.setItem('chosenPicture', this.state.chosenPicture);
    localStorage.setItem('chosenSound', this.state.chosenSound);
    localStorage.setItem('chosenText', this.state.chosenText);
    console.log("Numbers transmitted to local storage");
    alert("Art saved, you can press 'Fetch Art' anytime to see your saved art")
  }

  handleFetch = () => {
    this.setState({chosenPicture: Number(localStorage.getItem('chosenPicture'))});
    this.setState({chosenSound: Number(localStorage.getItem('chosenSound'))});
    this.setState({chosenText: Number(localStorage.getItem('chosenText'))});
    alert("Art fetched, check your tabs to se your saved art")
  }

  //upon clicking "generate art"
  handleClick = (formSubmitEvent) => {
    formSubmitEvent.preventDefault();
    console.log("handleClick clicked!");
    alert("Art generated! Check your tabs to see your collection")
  }

  handleClear = () => {
    //Reset state:
    this.setState({chosenText: this.null, chosenSound: this.null, chosenPicture: this.null});
    alert("Art cleared, click a new tab to update")
  };

  whichTab = (tab) => {
    this.setState ({
      activeTab: tab
    })
  }
  

  handleOptionChange = changeEvent => {
      if((Number(changeEvent.target.value))<4){
        this.setState({chosenPicture: Number(changeEvent.target.value)})
      }
      else if((Number(changeEvent.target.value))>3 && (Number(changeEvent.target.value))<7){
        this.setState({chosenSound: Number(changeEvent.target.value)})
      }
      else if((Number(changeEvent.target.value))>6){
        this.setState({chosenText: Number(changeEvent.target.value)})
      }
  };

render(){  
  return(
    console.log("App starting..."),
    <div className="App">
      <Header/>
      <div className="container1">
        
        <Tabs className="Tabs" whichTab={this.whichTab}>
        <div label="Welcome" className="mediaContent">
            <div className='image1'><img src={kiwi} className="kiwi-bird" alt="kiwi" /></div>
            <div className='text1'><h5>Create your own Gallery</h5><p>1. Choose one category for picture, sound and text </p>
            <p>2. Press "Generate art"</p><p>3. Click on the tabs to see your collection</p><p>4. Click "Save art" to save the collection</p></div> 
        </div>
        <div label="1" classname="mediaContent">
          <Page soundNr={this.state.chosenSound} pictureNr={this.state.chosenPicture} textNr={this.state.chosenText} activeTab={this.state.activeTab}/>    
        </div>
        <div label="2" classname="mediaContent">
          <Page soundNr={this.state.chosenSound} pictureNr={this.state.chosenPicture} textNr={this.state.chosenText} activeTab={this.state.activeTab}/>
        </div>
        <div label="3" classname="mediaContent">
          <Page soundNr={this.state.chosenSound} pictureNr={this.state.chosenPicture} textNr={this.state.chosenText} activeTab={this.state.activeTab}/>
        </div>
        <div label="4" classname="mediaContent">
          <Page soundNr={this.state.chosenSound} pictureNr={this.state.chosenPicture} textNr={this.state.chosenText} activeTab={this.state.activeTab}/> 
        </div>
        </Tabs> 
        
        <div className= "checkboxesStyle">
          <h2>PICTURE</h2>
          <form>
          <div className="pictures">
          <label>
              <input type="radio" name="picturesGroup" value="1" className="form-check-input" checked={this.state.chosenPicture === 1} onChange={this.handleOptionChange}/>
              Elephant
            </label>
          </div>
          <div className="pictures">
            <label>
              <input type="radio" name="picturesGroup" value="2" className="form-check-input" checked={this.state.chosenPicture === 2} onChange={this.handleOptionChange}/>
              Giraffe
            </label>
          </div>
          <div className="pictures">
            <label>
              <input type="radio" name="picturesGroup" value="3" className="form-check-input" checked={this.state.chosenPicture === 3} onChange={this.handleOptionChange}/>
              Dog
            </label>
          </div>
      </form>
          <h2>SOUNDS</h2>
          <form>
          <div className="sounds">
          <label>
              <input type="radio" name="soundsGroup" value="4" className="form-check-input-2" checked={this.state.chosenSound === 4} onChange={this.handleOptionChange}/>
              Techno
            </label>
          </div>
          <div className="sounds">
            <label>
              <input type="radio" name="soundsGroup" value="5" className="form-check-input-2" checked={this.state.chosenSound === 5} onChange={this.handleOptionChange}/>
              Jazz
            </label>
          </div>
          <div className="sounds">
            <label>
              <input type="radio" name="soundsGroup" value="6" className="form-check-input-2" checked={this.state.chosenSound === 6} onChange={this.handleOptionChange}/>
              Ringtone
            </label>
          </div>
      </form>
          <h2>TEXT</h2>
          <form>
          <div className="text">
          <label>
              <input type="radio" name="textGroup" value="7" className="form-check-input-3" checked={this.state.chosenText === 7} onChange={this.handleOptionChange}/>
              Poem
            </label>
          </div>
          <div className="text">
            <label>
              <input type="radio" name="textGroup" value="8" className="form-check-input-3" checked={this.state.chosenText === 8} onChange={this.handleOptionChange}/>
              Novel
            </label>
          </div>
          <div className="text">
            <label>
              <input type="radio" name="textGroup" value="9" className="form-check-input-3" checked={this.state.chosenText === 9} onChange={this.handleOptionChange}/>
              Lyrics
            </label>
          </div>
      </form>
      </div>
      </div>    
      <div className='container2'>
        <div onClick={this.handleSave} className="buttonStyle">SAVE ART</div>
        <div onClick={this.handleFetch} className="buttonStyle">FETCH ART</div>
        <div onClick={this.handleClear} className="buttonStyle">CLEAR ART</div>
        <div onClick={this.handleClick} type="submit" className="buttonStyle">GENERATE ART</div>
      </div>     
    </div>
  );
}};
export default App;