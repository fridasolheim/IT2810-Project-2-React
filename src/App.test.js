import React from 'react';
import App from './App'
import Header from './Components/Headerw'
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree= renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();

})
 /* const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});*/



/*import React from 'react';
import Link from '../Link.react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Link page="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});*/
