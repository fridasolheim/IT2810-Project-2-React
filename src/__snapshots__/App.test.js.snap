// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <span
      className="firstHeader"
      style={
        Object {
          "fontSize": "23px",
        }
      }
    >
      WELCOME 
    </span>
    <span
      className="secondHeader"
      style={
        Object {
          "fontSize": "13px",
        }
      }
    >
      TO OUR RANDOM ART GENERATOR
    </span>
    <div
      className="headerForm"
    >
      <form
        style={Object {}}
      >
        <div>
          <input
            className="inputName"
            onChange={[Function]}
            placeholder="What's your name"
            type="text"
          />
          <div
            className="headerSubmit"
            onClick={[Function]}
          >
            SUBMIT
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="container1"
  >
    <div
      className="tabs"
    >
      <ol
        className="tab-list"
      >
        <li
          className="tab-list-item tab-list-active"
          onClick={[Function]}
        >
          Welcome
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          1
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          2
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          3
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          4
        </li>
      </ol>
      <div
        className="tab-content"
      >
        <div
          className="image1"
        >
          <img
            alt="kiwi"
            className="kiwi-bird"
            src="kiwi.svg"
          />
        </div>
        <div
          className="text1"
        >
          <h5>
            Create your own Gallery
          </h5>
          <p>
            1. Choose one category for picture, sound and text 
          </p>
          <p>
            2. Press "Generate art"
          </p>
          <p>
            3. Click on the tabs to see your collection
          </p>
          <p>
            4. Click "Save art" to save the collection
          </p>
        </div>
      </div>
    </div>
    <div
      className="checkboxesStyle"
    >
      <h2>
        PICTURE
      </h2>
      <form>
        <div
          className="pictures"
        >
          <label>
            <input
              checked={true}
              className="form-check-input"
              name="picturesGroup"
              onChange={[Function]}
              type="radio"
              value="1"
            />
            Elephant
          </label>
        </div>
        <div
          className="pictures"
        >
          <label>
            <input
              checked={false}
              className="form-check-input"
              name="picturesGroup"
              onChange={[Function]}
              type="radio"
              value="2"
            />
            Giraffe
          </label>
        </div>
        <div
          className="pictures"
        >
          <label>
            <input
              checked={false}
              className="form-check-input"
              name="picturesGroup"
              onChange={[Function]}
              type="radio"
              value="3"
            />
            Dog
          </label>
        </div>
      </form>
      <h2>
        SOUNDS
      </h2>
      <form>
        <div
          className="sounds"
        >
          <label>
            <input
              checked={true}
              className="form-check-input-2"
              name="soundsGroup"
              onChange={[Function]}
              type="radio"
              value="4"
            />
            Techno
          </label>
        </div>
        <div
          className="sounds"
        >
          <label>
            <input
              checked={false}
              className="form-check-input-2"
              name="soundsGroup"
              onChange={[Function]}
              type="radio"
              value="5"
            />
            Jazz
          </label>
        </div>
        <div
          className="sounds"
        >
          <label>
            <input
              checked={false}
              className="form-check-input-2"
              name="soundsGroup"
              onChange={[Function]}
              type="radio"
              value="6"
            />
            Ringtone
          </label>
        </div>
      </form>
      <h2>
        TEXT
      </h2>
      <form>
        <div
          className="text"
        >
          <label>
            <input
              checked={true}
              className="form-check-input-3"
              name="textGroup"
              onChange={[Function]}
              type="radio"
              value="7"
            />
            Poem
          </label>
        </div>
        <div
          className="text"
        >
          <label>
            <input
              checked={false}
              className="form-check-input-3"
              name="textGroup"
              onChange={[Function]}
              type="radio"
              value="8"
            />
            Novel
          </label>
        </div>
        <div
          className="text"
        >
          <label>
            <input
              checked={false}
              className="form-check-input-3"
              name="textGroup"
              onChange={[Function]}
              type="radio"
              value="9"
            />
            Lyrics
          </label>
        </div>
      </form>
    </div>
  </div>
  <div
    className="container2"
  >
    <div
      className="buttonStyle"
      onClick={[Function]}
    >
      SAVE ART
    </div>
    <div
      className="buttonStyle"
      onClick={[Function]}
    >
      FETCH ART
    </div>
    <div
      className="buttonStyle"
      onClick={[Function]}
    >
      CLEAR ART
    </div>
    <div
      className="buttonStyle"
      onClick={[Function]}
      type="submit"
    >
      GENERATE ART
    </div>
  </div>
</div>
`;
